core/src/transport/memory.rs

```diff
@@ -245,51 +316,17 @@ impl error::Error for MemoryTransportError {}
 
 /// Listener for memory connections.
 pub struct Listener {
+    id: ListenerId,
     /// Port we're listening on.
     port: NonZeroU64,
     /// The address we are listening on.
     addr: Multiaddr,
     /// Receives incoming connections.
     receiver: ChannelReceiver,
-    /// Generate `ListenerEvent::NewAddress` to inform about our listen address.
+    /// Generate `TransportEvent::NewAddress` to inform about our listen address.
```

Comment:
```suggestion
    /// Generate [`TransportEvent::NewAddress`] to inform about our listen address.
```

Throws a docs compile error in case we ever rename this event.